codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/SyncClientClass.java

```diff
@@ -390,4 +404,19 @@ private MethodSpec waiterMethod() {
                          .returns(poetExtensions.getSyncWaiterInterface())
                          .build();
     }
+
+    private MethodSpec batchMangerMethod() {
+        String scheduledExecutor = "executorService";
+
+        BatchManagerMethod config = model.getCustomizationConfig().getBatchManagerMethod();
+        ClassName returnType = PoetUtils.classNameFromFqcn(config.getReturnType());
+
+        return MethodSpec.methodBuilder(BatchManagerMethod.METHOD_NAME)
+                         .returns(returnType)
+                         .addModifiers(Modifier.PUBLIC)
+                         .addAnnotation(Override.class)
+                         .addStatement("return $T.builder().client(this).executor($N).scheduledExecutor($N).build()",
+                                       returnType, scheduledExecutor, scheduledExecutor)
+                         .build();
+    }
```

Comment:
This looks very similar to the one in async client class. Can we reuse it?

Reply:
Reused the code by moving method into `ClientClassUtils`.